pkg/rules/slp115.go

```
@@ -0,0 +1,133 @@
+var slp115ExtensionGroups = []struct {
+	narrow  string
+	broader []string
+}{
+	{narrow: ".js", broader: []string{".js", ".mjs", ".cjs"}},
+	{narrow: ".ts", broader: []string{".ts", ".mts", ".cts"}},
+	{narrow: ".py", broader: []string{".py", ".pyi", ".pyw"}},
+	{narrow: ".css", broader: []string{".css", ".scss", ".less", ".sass"}},
+}
+
+func slp115IsExtBorder(b byte) bool {
+	return !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || (b >= '0' && b <= '9') || b == '_')
```

golangci-lint (2.11.4)

[error] QF1001: could apply De Morgan's law

(staticcheck)